### PR TITLE
Proved that both [,] and map respect function equality in Data.Sum.Properties

### PR DESCRIPTION
**Pull request by MirceaS**

### CODE REVIEW

**Comment by gallais** [comment]
In both `Data.List.Properties` and `Data.Maybe.Properties`, the same properties
for `map` are called `map-cong`. We should stick to that name (and adopt the same
naming convention for `[,]`.

**Comment by MirceaS** [comment]
Addressed comment

File: src/Data/Sum/Properties.agda [hunk 1]
```diff
@@ -44,6 +44,10 @@ module _ (dec₁ : Decidable {A = A} {B = A} _≡_)
 swap-involutive : swap {A = A} {B = B} ∘ swap ≗ id
 swap-involutive = [ (λ _ → refl) , (λ _ → refl) ]
 
+map-id : map (id {A = A}) (id {A = B}) ≗ id
```
Comment (MatthewDaggitt) [inline src/Data/Sum/Properties.agda]: You could get rid of half the brackets via `map {A = A} {B = B} id id ≗ id`?

**Comment by MirceaS** [comment]
I reckon this should be ready to be merged

**Comment by MirceaS** [comment]
Having used `[,]-∘-distr` extensively I found that it's only on very trivial cases that Agda can infer the value of the argument `f`, forcing me to always give it explicitly. Given that, I thought it would be a good idea to make the argument explicit to begin with.

File: src/Data/Sum/Properties.agda [hunk 2]
```diff
@@ -58,6 +62,22 @@ swap-involutive = [ (λ _ → refl) , (λ _ → refl) ]
 
 map-commute : {f : A → B}  {g : C → D}
               {f′ : B → E} {g′ : D → F} →
-              ((map f′ g′) ∘ (map f g)) ≗ map (f′ ∘ f) (g′ ∘ g)
+              (map f′ g′) ∘ (map f g) ≗ map (f′ ∘ f) (g′ ∘ g)
```
Comment (MirceaS) [inline src/Data/Sum/Properties.agda]: Those brackets were pointless so I removed them

Comment (gallais) [reply src/Data/Sum/Properties.agda]: The ones around `map` probably are too.

**Pull request merged**